compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUOps.td

```diff
@@ -289,66 +296,89 @@ def IREEGPU_CoalescedGatherDMAOp : Op<IREEGPU_Dialect, "coalesced_gather_dma", [
 
     ## Operands and Results
 
-    * `$indices`: Tensor/memref of index type specifying gather locations in
-      the source tensor
-    * `$source`: Source tensor/memref containing the data to be gathered
+    * `$indices`: The variadic `indices` operand is an optional tensor or vector of indices
+      to gather from. If the indices are present, their shape must be a prefix of
+      the `init`/`result` type.
+
+      The values in `indices` are linear indices into the memory starting at
+      `source` from which a given thread will gather data, and each tensor/vector component in `indices`
+      corresponds to one index dimension in `source`.
+      When `indices` has missing dimensions (or, in general, is fully absent), the index vector is
+      implicitly constructed as:
+      ```
+      realIndices[i0, i1, ..., iL, b0, b1, ..., bK] =
+        indices[i0, i1, ..., iL] + linear_index([b0, b1, ..., bK], [missing dimension sizes])
+      ```
+      That is, the missing index values form a contiguous slice starting at `index`.
+
+      This also means that, if no indices are specified, the tensor/memref at
+      `source` will be copied into `result` and must be contiguous in memory.
+
+      Once the `realIndices` have been expanded, this operation will gather data
+      into its result as by setting:
+      ```
+      result[i0, i1, ..., iN + lane_id * contiguous_length] = source[realIndices[i0, i1, ..., iN]]
```

Comment:
This is now `result[i0, i1, ... iN-1, (iN + lane_id * dim(source, N)] = source[indices[0][i0], indices[1][i1], ..., indices[N][iN]]` (including the implicit dimensions thing)